examples/systemcalls/systemcalls.c

```diff
@@ -16,6 +22,15 @@ bool do_system(const char *cmd)
  *   and return a boolean true if the system() call completed with success
  *   or false() if it returned a failure
 */
+    int res = 0;
+
+    res = system(cmd);
+    
+    if(-1 == res)
+    {   
+        printf("\nError: system() returned error. Error code: %d", errno); 
+        return false;
```

Comment:
Is this check sufficient?  I think there are cases where system would return a value other than -1 here and you need to return false.